src/metabase/db/liquibase.clj

```diff
@@ -119,7 +139,7 @@
     (if (has-unrun-migrations? liquibase)
       (do
         (log/info (trs "Migration lock is cleared. Running migrations..."))
-        (u/auto-retry 3 (let [^Contexts contexts nil] (.update liquibase contexts))))
```

Comment:
Just curious, why remove the retries here?

Reply:
It's more trouble than it's worth I think... the problem is sometimes migrations fail halfway through a single changeSet and then instead of showing the actual error it tries to run the changeSet again and shows a "table ____ already exists" error. So it tends to obscure the actual errors and actually makes things a lot more confusing.

(This is the reason why we restrict changeSets to a single change each now, but a lot of our older migrations have several changes per change set, meaning they're not atomic in DBs like MySQL that can't do DDL inside of transactions)

Reply:
I ran in to this while debugging some of the tweaks I had to make for MySQL/MariaDB -- errors were super hard to debug because it didn't display the actual error message. If running a migration fails it's not really likely to succeed if you retry it anyway